Framework/src/InfrastructureSpecReader.cxx

```diff
@@ -93,7 +93,14 @@ TaskSpec InfrastructureSpecReader::readSpecEntry<TaskSpec>(const std::string& ta
   ts.className = taskTree.get<std::string>("className");
   ts.moduleName = taskTree.get<std::string>("moduleName");
   ts.detectorName = taskTree.get<std::string>("detectorName");
-  ts.cycleDurationSeconds = taskTree.get<int>("cycleDurationSeconds");
+  ts.cycleDurationSeconds = taskTree.get<int>("cycleDurationSeconds", -1);
+  if (taskTree.count("cycleDurations") > 0) {
+    for (const auto& cycleConfig : taskTree.get_child("cycleDurations")) {
+      auto duration = cycleConfig.second.get<size_t>("cycleDuration");
+      auto nbCycles = cycleConfig.second.get<size_t>("period");
+      ts.cycleDurations.push_back(std::pair{ duration, nbCycles });
```

Comment:
didn't you decide to use period/validity instead of `nbCycles`? I am confused now

Reply:
yes that was a leftover